src/refactorings/change-signature/change-signature.ts

```diff
@@ -0,0 +1,198 @@
+import { Editor, ErrorReason, SelectedPosition } from "../../editor/editor";
+import { Selection } from "../../editor/selection";
+import * as t from "../../ast";
+import { Path } from "../../editor/path";
+
+let toModifyNode: t.FunctionDeclaration;
+
+export async function changeSignature(editor: Editor) {
+  const params =
+    toModifyNode?.params.map((p, index) => {
+      return {
+        label: getParamName(p),
+        value: {
+          startAt: index,
+          endAt: index
+        }
+      };
+    }) || [];
+
+  await editor.askForPositions(params, async (message) => {
+    const { selection } = editor;
+    const refrences = await editor.getSelectionReferences(selection);
+
+    const filesContent = await Promise.all(
+      refrences.map(async (reference) => {
+        const content = await editor.codeOf(reference.path);
+        return {
+          code: content,
+          path: reference.path,
+          selection: reference.selection
+        };
+      })
+    );
+
+    const alreadyTransformed: Record<string, string> = {};
+    const result: {
+      path: Path;
+      transformed: t.Transformed;
+    }[] = [];
+
+    for (const x of filesContent) {
+      const codeToTransform =
+        alreadyTransformed[x.path.value] || (x.code as string);
+
+      try {
+        const transformed = updateCode(
+          t.parse(codeToTransform),
+          x.selection,
+          message
+        );
+
+        alreadyTransformed[x.path.value] = `${transformed.code}`;
+
+        result.push({
+          path: x.path,
+          transformed
+        });
+      } catch (error) {
```

Comment:
🍰 Could we log this error in the console at least? That could help debugging.

Reply:
console.log added in https://github.com/nicoespeon/abracadabra/pull/725/commits/bea756d2a38c293388e5ade61c39ed77c1dffdc1